src/clients/ApplicationClient.js

```diff
@@ -704,7 +704,130 @@ export default class ApplicationClient extends BaseClient {
+    this.log.debug("[ApplicationClient] removeDevicesFromGroup("+groupId+","+deviceList+")");
+    return this.callApi('PUT', 200, false, ['bulk', 'devices' , groupId, "remove"], deviceList);
+  }
+
+  // https://console.ng.bluemix.net/docs/services/IoT/gateways/gateway-access-control.html
+  
+	// Finding a Resource Group
+		// getGatewayGroups()
+	// Querying a resource group
+		// getUniqueDevicesInGroup(groupId)
+		// getUniqueGatewayGroup(groupId)
+	// Creating and deleting a resource group
+		// createGatewayGroup(groupName)
+		// deleteGatewayGroup(groupId)
+	// Retrieving and updating device properties
+		// getGatewayGroupProperties()
+		// getDeviceRoles(deviceId)
+		// updateGatewayProperties(gatewayId,control_props)
+		// updateDeviceControlProperties(deviceId, withroles)
+
+  // Finding a Resource Group
+  getGatewayGroups(){
+    this.log.debug("[ApplicationClient] getGatewayGroups()");
+    return this.callApi('GET', 200, true, ['groups'], null);  
+   }
+   
+  // Querying a resource group
+
+  // Get unique identifiers of the members of the resource group
+  getUniqueDevicesInGroup(groupId){
+    this.log.debug("[ApplicationClient] getUniqueDevicesInGroup("+groupId+")");
+    return this.callApi('GET', 200, true, ['bulk', 'devices' , groupId, "ids"], null);
+  }
+
+  // properties of the resource group
+  getUniqueGatewayGroup(groupId){
+    this.log.debug("[ApplicationClient] getUniqueGatewayGroup("+groupId+")");
+    return this.callApi('GET', 200, true, ['groups', groupId], null);
+  } 
+  
+  // Creating and deleting a resource group
+
+  // Create a Resource Group
+  createGatewayGroup(groupName){
+    this.log.debug("[ApplicationClient] createGatewayGroup()");
+    return this.callApi('POST', 201, true, ['groups'], groupName);
+  } 
+  
+  // Delete a Resource Group
+  deleteGatewayGroup(groupId){
+    this.log.debug("[ApplicationClient] deleteGatewayGroup("+groupId+")");
+    return this.callApi('DELETE', 200, false, ['groups', groupId], null);
+  }
+
+  // Retrieving and updating device properties
+  
+  // Get the ID of the devices group of a gateway
+  getGatewayGroupProperties(){
+    this.log.debug("[ApplicationClient] getGatewayGroupProperties()");
+    return this.callApi('GET', 200, true, ['authorization', 'devices' ], null);
+  }
+
+  // Get standard role of a gateway
+  getDeviceRoles(deviceId){
+    this.log.debug("[ApplicationClient] getDeviceRoles("+deviceId+")");
+    return this.callApi('GET', 200, true, ['authorization', 'devices', deviceId, 'roles'], null);
+  }  
+
+  // Update device properties without affecting the access control properties
+  updateGatewayProperties(gatewayId,control_props){
+    this.log.debug("[ApplicationClient] updateGatewayProperties("+gatewayId+")");
+    return this.callApi('PUT', 200, true, ['authorization', 'devices' , gatewayId], control_props);
+  }
+  
+  // Assign a standard role to a gateway
+  updateDeviceControlProperties(deviceId, withroles){
```

Comment:
can we perhaps start breaking the api down a little rather being a flat api ...

client = new iotmagicclient(...)

mgr = client.getAccessControlManager().
mgr.updateDevice(....)

registry = client.getDeviceRegistryManager().
registry.updateDevice(....)

it's a pretty standard model

I'd expect the breakdown to map closely to the top level api sections on the site under neath api/v0002/XXX

e.g. logs
device
mgmt
etc
etc

I've been working on refactoring the python library in this kind of manner on the side, i think it's a much better way to structure a client library than just a client with 100 methods